### PR TITLE
[data, model] fix: coalesce qwen3.5 pad cu-seqlens for GDN

### PR DESCRIPTION
**Pull request by YLlllllllllll**
### What does this PR do?

Fix Qwen3.5/Qwen3.5-MoE `pad_to_length=true` with sequence parallelism for the Gated DeltaNet path.

The previous `cu_seqlens` construction could contain duplicated tail-padding boundaries after Ulysses/SP chunking. FlashAttention can tolerate zero-length or padding-only packed segments, but the GDN/linear-attention path forwards those boundaries into FLA/FlashQLA kernels. This PR coalesces known tail-padding segments for linear-attention kernels while preserving the original FlashAttention kwargs.

### Checklist Before Starting

- Search for relative PRs/issues and link here: internal Codebase MR https://code.byted.org/data/Open-VeOmni/merge_requests/167
- PR title follows `[{modules}] {type}: {description}` format.

### Test

- `git diff --check origin/main..HEAD` passed.
- `python3 -m py_compile veomni/data/data_collator.py tests/data/test_collators.py tests/data/test_prepare_fa_kwargs.py` passed.
- `PYTHONPATH=/home/tiger/modelchef/.venv/lib/python3.11/site-packages python3 -m pytest tests/data/test_prepare_fa_kwargs.py tests/data/test_collators.py`: 11 passed.
- H20 2x8, Qwen3.5-9B, seq=131072, Ulysses/SP=2, `pad_to_length=true`: FlashQLA passed 120 steps and FLA completed 150/150 steps in the same JobRun.
- A100 1x8 FLA 200-step serial comparison: `pad_to_length=true` preserved training loss and reduced step-time/GPU-utilization variance.

Evidence doc: https://bytedance.larkoffice.com/docx/YOBjd4DdGopuFox4G3gczvhDndc

### API and Usage Example

No user-facing API change. Existing `--train.pad_to_length true` now works for Qwen3.5/Qwen3.5-MoE GDN paths under sequence parallelism.

### Design & Code Changes

- Add a utility to coalesce duplicated tail-padding `cu_seqlens` boundaries for linear-attention/GDN packed inputs.
- Compute linear-attention tail padding from the actual `input_ids` length delta before/after `pad_batch_to_length`, so bool-style `pad_to_length` and future alignment padding cannot misreport the tail length.
- Apply the coalesced boundaries only to the Qwen3.5/Qwen3.5-MoE GDN path.
- Keep FlashAttention kwargs compatible with the existing packed-input contract.
- Regenerate patched Qwen3.5/Qwen3.5-MoE model code and patch diffs.
- Extend data/packing tests for duplicated tail-padding and actual coalesced `linear_attn_cu_seq_lens_q` cases.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

File: veomni/data/data_collator.py [hunk 1]
```diff
@@ -367,12 +389,14 @@ def __call__(self, batch: Dict[str, Union[torch.Tensor, List[torch.Tensor]]]) ->
                 post_pad_len = len(batch[key]) if isinstance(batch[key], list) else batch[key].size(pack_dim)
                 if key in vit_sp_pad:
                     vit_sp_pad[key] = post_pad_len - pre_pad_len
+                if key == "position_ids":
+                    linear_attn_tail_padding_length += post_pad_len - pre_pad_len
```
Comment (Coach257) [inline veomni/data/data_collator.py:393]: please use `valid_seqlens_from_cu_seqlens` to simplify the code.

Comment (YLlllllllllll) [reply veomni/data/data_collator.py:393]: Updated in the latest push.

`coalesce_tail_padding_cu_seqlens` now reuses `valid_seqlens_from_cu_seqlens` via an exact `tail_padding_length` argument, and I added a unit test for the edge case where the final real packed sample has length 1.

I kept the explicit SP padding delta in the collator because inferring all trailing `+1` segments from `cu_seqlens` alone is not equivalent: for `cu=[0,4,5,6,7,8]` with real lengths `[4,1]` and tail padding `3`, heuristic inference would return `[4]`, while the exact tail length correctly preserves `[4,1]` and coalesces to `[0,4,5,8]`.

**Review by Luosuu — APPROVED** [review]

**Comment by Luosuu** [comment]
@YLlllllllllll please fix CI

**Comment by YLlllllllllll** [comment]
@Luosuu fixed the CI lint issue and pushed a new head `f30e2214`.

Root cause was only `ruff format --check` on `veomni/utils/seqlen_pos_transform_utils.py`; GPU/NPU unit/e2e jobs were skipped behind the failed lint job. I reran locally:

- `.venv/bin/python -m ruff check tasks tests veomni docs && .venv/bin/python -m ruff format --check tasks tests veomni docs`
- `python3 -m py_compile veomni/data/data_collator.py veomni/utils/seqlen_pos_transform_utils.py tests/data/test_collators.py tests/data/test_prepare_fa_kwargs.py`
- `PYTHONPATH=/home/tiger/modelchef/.venv/lib/python3.11/site-packages python3 -m pytest tests/data/test_prepare_fa_kwargs.py tests/data/test_collators.py` -> 11 passed

The new GitHub Actions runs are currently `action_required` with no jobs created; GitHub API says approving fork workflow runs requires repository admin rights, so a maintainer needs to approve/run the checks.
